### PR TITLE
refactor: extract runtime config to releases.exs

### LINKED ISSUE #31: Clean up config
At the moment we have a lot of config sitting in here:
https://github.com/supabase/realtime/blob/1d7cf1162eb40e5f3c6c9349e2509cd7f7f8914c/server/lib/realtime/application.ex#L13

This was a quick hack, and I think it can be refactored out to the config folder so that it follows normal convention and is a bit cleaner:
https://github.com/supabase/realtime/tree/master/server/config

### PR DESCRIPTION
**Pull request by soedirgo**
This will probably be affected by #48 in the future, but for now we should follow the recommended way to set runtime configs as explained [here](https://hexdocs.pm/mix/1.9.0/Mix.Tasks.Release.html#module-runtime-configuration).

Closes #31.

### CODE REVIEW

File: server/config/releases.exs [hunk 1]
```diff
@@ -1,28 +1,35 @@
 import Config
 
-app_port = System.get_env("PORT") || 4000
 app_hostname = System.get_env("HOSTNAME") || "localhost"
-db_user = System.get_env("DB_USER") || "postgres"
-db_password = System.get_env("DB_PASSWORD") || "postgres"
+app_port = System.get_env("PORT") || "4000"
 db_host = System.get_env("DB_HOST") || "localhost"
-db_port = System.get_env("DB_PORT") || 5432
+db_port = System.get_env("DB_PORT") || "5432"
```
Comment (kiwicopple) [inline server/config/releases.exs:6]: Should this one be a number @soedirgo ?

File: server/config/releases.exs [hunk 2]
```diff
@@ -1,28 +1,35 @@
 import Config
 
-app_port = System.get_env("PORT") || 4000
 app_hostname = System.get_env("HOSTNAME") || "localhost"
-db_user = System.get_env("DB_USER") || "postgres"
-db_password = System.get_env("DB_PASSWORD") || "postgres"
+app_port = System.get_env("PORT") || "4000"
```
Comment (kiwicopple) [inline server/config/releases.exs:4]: Should this one be a number?

Comment (soedirgo) [reply server/config/releases.exs:4]: Seems to crash with a number:
```
ERROR! Config provider Config.Reader failed with:
** (ArgumentError) argument error
    :erlang.binary_to_integer(4000)
    (stdlib 3.13) erl_eval.erl:680: :erl_eval.do_apply/6
    (stdlib 3.13) erl_eval.erl:888: :erl_eval.expr_list/6
    (stdlib 3.13) erl_eval.erl:240: :erl_eval.expr/5
    (stdlib 3.13) erl_eval.erl:232: :erl_eval.expr/5
    (stdlib 3.13) erl_eval.erl:233: :erl_eval.expr/5
    (stdlib 3.13) erl_eval.erl:888: :erl_eval.expr_list/6
    (stdlib 3.13) erl_eval.erl:240: :erl_eval.expr/5

{"init terminating in do_boot",{badarg,[{erlang,binary_to_integer,[4000],[]},{erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,680}]},{erl_eval,expr_list,6,[{file,"erl_eval.erl"},{line,888}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,240}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,232}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,233}]},{erl_eval,expr_list,6,[{file,"erl_eval.erl"},{line,888}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,240}]}]}}
init terminating in do_boot ({badarg,[{erlang,binary_to_integer,[4000],[]},{erl_eval,do_apply,6,[{_},{_}]},{erl_eval,expr_list,6,[{_},{_}]},{erl_eval,expr,5,[{_},{_}]},{erl_eval,expr,5,[{_},{_}]},{erl

Crash dump is being written to: erl_crash.dump...done
```

Works fine with a string though (the build fail is something different, looking into it)

Comment (kiwicopple) [reply server/config/releases.exs]: Cool 👍

**Pull request merged**